### PR TITLE
[AIFlow] Airflow support scheduling base on TASK_STATUS_CHANGE event

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Airflow support scheduling base on TASK_STATUS_CHANGE event

## Brief change log
- Send TASK_STATUS_CHANGE event when airflow task state is changed
- Use EventHandlerManager to handle TASK_STATUS_CHANGE

## Verifying this change

This change added tests.

### CODE REVIEW

File: flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py [hunk 1]
```diff
@@ -221,12 +223,13 @@ def from_base_event(cls, event: BaseEvent) -> 'SchedulerInnerEvent':
 
 class TaskStatusChangedEvent(SchedulerInnerEvent):
```
Comment (SteNicholas) [inline flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py:224]:
```suggestion
class TaskStateChangedEvent(SchedulerInnerEvent):
```

File: flink-ai-flow/ai_flow/api/ops.py [hunk 2]
```diff
@@ -518,21 +518,24 @@ def action_on_dataset_event(job_name: Text,
...
-                         action: TaskAction = TaskAction.START):
+                         action: TaskAction = TaskAction.START,
+                         condition_type: ConditionType = ConditionType.SUFFICIENT):
     """
     Trigger job by upstream job status changed.
     :param job_name: The job name
     :param upstream_job_name: The upstream job name
     :param upstream_job_status: The upstream job status, type: ai_flow.workflow.status.Status
     :param action: The ai_flow.workflow.control_edge.TaskAction type.
+    :param condition: The event condition. Sufficient or Necessary.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/api/ops.py:529]:
```suggestion
    :param condition: The condition. Sufficient or Necessary.
```

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py [hunk 3]
```diff
@@ -221,12 +223,13 @@ def from_base_event(cls, event: BaseEvent) -> 'SchedulerInnerEvent':
...
 
-    def __init__(self, task_id, dag_id, execution_date, status):
+    def __init__(self, task_id, dag_id, execution_date, status, create_time=None):
```
[the remark this answers is not in the document]
Comment (Sxnan) [reply flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py]: fixed in 4747574

File: flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py [hunk 4]
```diff
@@ -221,12 +223,13 @@ def from_base_event(cls, event: BaseEvent) -> 'SchedulerInnerEvent':
...
         super().__init__()
         self.task_id = task_id
         self.dag_id = dag_id
         self.execution_date = execution_date
         self.status = status
```
[the remark this answers is not in the document]
Comment (Sxnan) [reply flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py]: fixed in 4747574

File: flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py (hunk 1 above)
Comment (Sxnan) [reply flink-ai-flow/lib/airflow/airflow/events/scheduler_events.py]: fixed in 4747574

File: flink-ai-flow/ai_flow/api/ops.py (hunk 2 above)
Comment (Sxnan) [reply flink-ai-flow/ai_flow/api/ops.py]: Fixed in 6610a98

File: flink-ai-flow/lib/airflow/airflow/contrib/jobs/event_based_scheduler_job.py [hunk 5]
```diff
@@ -140,10 +142,15 @@ def schedule(self):
...
-                        self.mailbox.send_message(DagRunFinishedEvent(dagrun.run_id).to_event())
+                    if dagrun is not None:
+                        dag_run_id = DagRunId(dagrun.dag_id, dagrun.run_id)
+                        self.task_event_manager.handle_event(dag_run_id, origin_event)
+                        tasks = self._find_scheduled_tasks(dagrun, session)
+                        self._send_scheduling_task_events(tasks, SchedulingAction.START)
+                        if dagrun.state in State.finished:
+                            self.mailbox.send_message(DagRunFinishedEvent(dagrun.run_id).to_event())
+                    else:
+                        self.log.error("dagrun is None for dag_id:{} execution_date: {}".format(event.dag_id, event.execution_date))
```
[the remark this answers is not in the document]
Comment (Sxnan) [reply flink-ai-flow/lib/airflow/airflow/contrib/jobs/event_based_scheduler_job.py]: Fixed in a7e1528

File: flink-ai-flow/lib/airflow/airflow/jobs/local_task_job.py [hunk 6]
```diff
@@ -134,6 +143,17 @@ def signal_handler(signum, frame):
...
             self.on_kill()
 
+    def _send_task_status_change_event(self):
+        task_status_changed_event = TaskStatusChangedEvent(
+            self.task_instance.task_id,
+            self.task_instance.dag_id,
+            self.task_instance.execution_date,
+            self.task_instance.state
+        )
+        event = task_status_changed_event.to_event()
```
[the remark this answers is not in the document]
Comment (Sxnan) [reply flink-ai-flow/lib/airflow/airflow/jobs/local_task_job.py:153]: fixed in a7e1528

**Review by SteNicholas — APPROVED** [review]

File: flink-ai-flow/lib/airflow/airflow/contrib/jobs/event_based_scheduler_job.py [hunk 7]
```diff
@@ -311,7 +313,7 @@ def _schedule_task(self, scheduling_event: TaskSchedulingEvent):
             scheduling_event.execution_date,
             scheduling_event.try_number
         )
-        self.executor.schedule_task(task_key, scheduling_event.action)
+        self.executor.schedule_task(task_key, scheduling_event.action, self.server_uri)
```
Comment (wuchaochen) [inline flink-ai-flow/lib/airflow/airflow/contrib/jobs/event_based_scheduler_job.py:321]: Change server_uri to the parameter of executor.

Comment (Sxnan) [reply flink-ai-flow/lib/airflow/airflow/contrib/jobs/event_based_scheduler_job.py]: Fixed in 4147015

**Review by wuchaochen — APPROVED** [review]

**Pull request merged**